src/output-json-alert.c

```diff
@@ -243,6 +243,10 @@ void AlertJsonHeader(const Packet *p, const PacketAlert *pa, json_t *js)
         }
     }
 
+    if (p->flow) {
+        json_object_set_new(js, "app_proto",
```

Comment:
2 thoughts:

1. perhaps it makes sense to turn the code here into a util func: https://github.com/inliniac/suricata/blob/master/src/output-json-flow.c#L191 and call that. That gives a bit more info in some sigs, like for the app-layer protocol detect warnings.

2. alternatively, what about adding the partial flow record to the alert record? Just like how we add http etc we can add the flow record. This would then automatically include the app_proto